### PR TITLE
libc: Move math library from libs/libc/math to libs/libm/libm

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

to prepare the support of other implementation e.g.:
https://github.com/JuliaMath/openlibm
https://gitlab.com/gtd-gmbh/libmcs

## Impact

split math code from libc to libm

## Testing

CI

please ignore the following false nxstyle waring:
```
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:126:51: error: Operator/assignment must be preceded with whitespace
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:139:51: error: Operator/assignment must be preceded with whitespace
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:162:51: error: Operator/assignment must be preceded with whitespace
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:166:51: error: Operator/assignment must be preceded with whitespace
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:167:51: error: Operator/assignment must be preceded with whitespace
Error: /home/runner/work/nuttx/nuttx/nuttx/libs/libm/libm/lib_lgamma.c:178:51: error: Operator/assignment must be preceded with whitespace
```

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@acassis this is the first step to support external libm implementation, please take a look.

**Comment by xiaoxiang781216** [comment]
@centurysys this patch could resolve your confusion happened in https://github.com/apache/nuttx/pull/8510.

**Comment by centurysys** [comment]
@xiaoxiang781216 
Thank you for doing all this hard work!

File: libs/libm/Kconfig [hunk 1]
```diff
@@ -3,11 +3,12 @@
 # see the file kconfig-language.txt in the NuttX tools repository.
 #
 
-#menu "Math Library Support"
+choice
+	prompt "Select math library"
+	default LIBM_NONE
```
Comment (pkarashchenko) [inline libs/libm/Kconfig:8]: Only one question to `default` value. Before selection was defaulted to what is called `LIBM_TOOLCHAIN` here, so maybe
```
default LIBM_NONE if DEFAULT_SMALL
default LIBM_TOOLCHAIN if !DEFAULT_SMALL
```
?

Comment (xiaoxiang781216) [reply libs/libm/Kconfig:8]: good idea

Comment (xiaoxiang781216) [reply libs/libm/Kconfig]: Done.

**Review by jerpelea — APPROVED** [review]

**Review by acassis — APPROVED** [review]

**Review by pkarashchenko — COMMENTED** [review]
Please fix RISC-V CI

**Comment by xiaoxiang781216** [comment]
Done.

**Review by jerpelea — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@pkarashchenko ci is finally pass, please merge it.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**